pkg/artifactory/resource/repository/remote/resource_artifactory_remote_conan_repository.go

```diff
@@ -0,0 +1,46 @@
+package remote
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jfrog/terraform-provider-artifactory/v6/pkg/artifactory/resource/repository"
+	"github.com/jfrog/terraform-provider-shared/packer"
+	"github.com/jfrog/terraform-provider-shared/util"
+)
+
+type ConanRemoteRepo struct {
+	RepositoryBaseParams
+	ForceConanAuthentication bool `json:"forceConanAuthentication"`
+}
+
+func ResourceArtifactoryRemoteConanRepository() *schema.Resource {
+	const packageType = "conan"
+
+	var conanRemoteSchema = util.MergeMaps(BaseRemoteRepoSchema, map[string]*schema.Schema{
+		"force_conan_authentication": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     false,
+			Description: `Force basic authentication credentials in order to use this repository. Default value is 'false'.`,
+		},
+	}, repository.RepoLayoutRefSchema("remote", packageType))
+
+	var unpackConanRemoteRepo = func(s *schema.ResourceData) (interface{}, string, error) {
```

Comment:
In this `func` definition, we are returning `error`. Do we really need that? At the end of the func definition, we are always returning `nil`.

Reply:
The code that makes use of this function has to allow for the possibility of errors. Technically, return results 1,2 can be collapsed and you could just return `Identifiable` because that's where `.Id()` comes from. Originally there wasn't a way to  supply struct and it's key as a result (hence, `repo`, `repo.Id()`).